src/app/public/testing/filter/filter-fixture.spec.ts

```diff
@@ -0,0 +1,176 @@
+   * components and the SKY UX modules that those components rely upon.
+   */
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [FilterTestComponent],
+      imports: [SkyFilterTestingModule]
+    });
+    fixture = TestBed.createComponent(FilterTestComponent);
+    testComponent = fixture.componentInstance;
+    filterSummaryFixture = new SkyFilterFixtureSummary(fixture, DATA_SKY_ID_SUMMARY);
+    filterButtonFixture = new SkyFilterFixtureButton(fixture, DATA_SKY_ID_BUTTON);
+  });
+
+  describe('Summary items', () => {
+    it('should dismiss only filter item from summary', async () => {
+      expect(testComponent.appliedFilters.length).toBe(0);
+      testComponent.applyFilter('example');
+      expect(testComponent.appliedFilters.length).toBeGreaterThan(0);
+      fixture.detectChanges();
+      await filterSummaryFixture.filterCloseClick(0);
+      fixture.detectChanges();
+      expect(testComponent.appliedFilters.length).toBe(0);
+    });
+
+    it('should dismiss nth filter item from summary', async () => {
+      expect(testComponent.appliedFilters.length).toBe(0);
+      for (let i = 1; i <= 5; i++) {
+        testComponent.applyFilter(`example ${i}`);
+      }
+      expect(testComponent.appliedFilters.length).toBe(5);
+      fixture.detectChanges();
+      await filterSummaryFixture.filterCloseClick(3);
+      fixture.detectChanges();
+      expect(testComponent.appliedFilters.length).toBe(4);
+    });
+
+    it('should throw error when dismissing non existent filter', async () => {
+      expect(testComponent.appliedFilters.length).toBe(0);
+      for (let i = 1; i <= 3; i++) {
+        testComponent.applyFilter(`example ${i}`);
+      }
+      expect(testComponent.appliedFilters.length).toBe(3);
+      fixture.detectChanges();
+      try {
+        await filterSummaryFixture.filterCloseClick(5);
+        expect('Error').toBeTrue();
+      } catch (e) {
+        expect(e).toBeInstanceOf(Error);
+        expect((e as Error).message).toBe('Unable to click close for a filter index 5');
+      }
+    });
+  });
+
+  describe('Button', () => {
+    it('should click the button', async () => {
+      expect(testComponent.buttonClicked).toBeFalse();
+      filterButtonFixture.clickFilterButton();
+      fixture.detectChanges();
+      expect(testComponent.buttonClicked).toBeTrue();
+    });
+
+    it('should do nothing when the button is not enabled', async () => {
+      testComponent.buttonIsDisabled = true;
+      expect(testComponent.buttonClicked).toBeFalse();
+      fixture.detectChanges();
+      await filterButtonFixture.clickFilterButton();
+      fixture.detectChanges();
+      expect(testComponent.buttonClicked).toBeFalse();
+    });
+
+    it('should get the button aria controls', async () => {
+      fixture.detectChanges();
+      await fixture.whenStable();
+      expect(filterButtonFixture.button.ariaControls).toBeFalsy();
```

Comment:
Are you concerned that checking for falsy might miss something? By default, `ariaControls` doesn't have a value and is therefore falsy. If this breaks, we might not know. Instead, could we set `[ariaControls]="'somevalue'"` in your fixture template, and then test for that value here? `expect(filterButtonFixture.button.ariaControls).toEqual('somevalue');`
I'd also check the same for `ariaExpanded` below ...